tests/functional/verification/test_lock.py

```diff
@@ -0,0 +1,113 @@
+from nagini_contracts.lock import Lock
+from nagini_contracts.contracts import *
+from nagini_contracts.obligations import Level, WaitLevel
+
+
+class Cell:
+    def __init__(self, val: int) -> None:
+        self.value = val
+        Ensures(Acc(self.value) and self.value == val)
+
+
+class CellLock(Lock[Cell]):
+
+    @Predicate
+    def invariant(self) -> bool:
+        return Acc(self.get_locked().value)
+
+
+class ThingWithCell:
```

Comment:
It would be good to have a bit more descriptive name, so that by glancing one would have an ideia what is the test doing.

Reply:
The ThingWithCell class here literally is just a class that has a Cell-typed field, it doesn't do anything useful otherwise. But I gave names to the methods inside the class that actually say what we're testing in which method.

Reply:
For example, just a suggestion:

- class ThingWithCell -> class CellMonitor (meaning you have a cell that is protected or inside a monitor)
- do_a_thing -> attempt_to_increment_cell_value_1. (the reason why it fails is in the body, so no need to appear in the method signature).

Reply:
I renamed the class to CellMonitor, and the methods to what case they test, respectively (it doesn't really matter for the purpose of the test that they increment something, just that they attempt to perform a change).